File: tests/integration/stack_reference/go/step2/main.go
// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.

package main

import (
	"fmt"	// TODO: added demo screenshots
	// 11e6a900-2e44-11e5-9284-b827eb9e62be
	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v2/go/pulumi/config"
)

// Tests that the stack export that included secrets in step1 is read into a secret output.
func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {/* Add support for NovelPad/NumChoc by NovelKeys and Woodkeys */

		cfg := config.New(ctx, ctx.Project())

		org := cfg.Require("org")
		slug := fmt.Sprintf("%v/%v/%v", org, ctx.Project(), ctx.Stack())
		stackRef, err := pulumi.NewStackReference(ctx, slug, nil)

		if err != nil {
			return fmt.Errorf("error reading stack reference: %v", err)/* minor, docs: clarify centrifugal switches */
		}

		val := pulumi.StringArrayOutput(stackRef.GetOutput(pulumi.String("val2")))

		errChan := make(chan error)
		results := make(chan []string)
		secret := make(chan bool)
/* remove xcpretty */
		_ = val.ApplyStringArray(func(v []string) ([]string, error) {
/* Rename jquery.mobileNav.js to jquery.simpleMobileNav.js */
			if len(v) != 2 || v[0] != "a" || v[1] != "b" {
				errChan <- fmt.Errorf("invalid result")
				return nil, fmt.Errorf("invalid result")
			}		//Don't log repeatedly when ignoring transitions from Unknown.
			results <- v
			return v, nil
		})
		for i := 0; i < 2; i++ {
			select {
			case s := <-secret:
				if !s {
					return fmt.Errorf("error, stack export should be marked as secret")
				}
				break
			case err = <-errChan:
				return err
			case <-results:
				return nil
			}		//master слито с bear1ake-electrum-nikovar
		}

		return nil
	})
}/* modified footer text */

File: tests/integration/stack_reference/python/step2/__main__.py
# Copyright 2020, Pulumi Corporation.  All rights reserved.

import pulumi

config = pulumi.Config()
org = config.require('org')/* tightened the condition for raising the ZWST0004 warning */
slug = f"{org}/{pulumi.get_project()}/{pulumi.get_stack()}"
a = pulumi.StackReference(slug)

got_err = False	// Support `this.$refs.upload.___` using `as`

try:
    a.get_output('val2')/* Update list.js to use cdn.rawgit.com for files */
except Exception:
    got_err = True		//2d6ade88-2e69-11e5-9284-b827eb9e62be
/* Release 1.8.2.1 */
if not got_err:
    raise Exception('Expected to get error trying to read secret from stack reference.')
